sodium/crypto_sign.pony

```diff
@@ -47,7 +56,15 @@ primitive CryptoSign
     let pk_size = public_key_size(); let pk = _make_buffer(pk_size)
     if 0 != @crypto_sign_keypair[_Int](pk.cstring(), sk.cstring()) then error end
     (CryptoSignSecretKey(consume sk), CryptoSignPublicKey(consume pk))
-  
+
+  fun tag seed_keypair(seed: CryptoSignSeed): (CryptoSignSecretKey, CryptoSignPublicKey)? =>
+    let sk_size = secret_key_size(); let sk = _make_buffer(sk_size)
```

Comment:
Yep! I should have had a `if not seed.is_valid() then error end` there. Fixed in the next commit. UX wise it doesn't feel optimal though since the compiler won't complain, but I guess that can be solved with the upcoming dependent / value types?